### ISSUE TITLE
Stop gradient for alpha in non alpha losses

### ISSUE THREAD

**Issue by jendelel** [rail-berkeley/softlearning#60]
Hi, 

in the [code](https://github.com/rail-berkeley/softlearning/blob/master/softlearning/algorithms/sac.py), you are propagating the gradient into alpha also from the [actor](https://github.com/rail-berkeley/softlearning/blob/master/softlearning/algorithms/sac.py#L301) and [critic losses](https://github.com/rail-berkeley/softlearning/blob/master/softlearning/algorithms/sac.py#L196). Is that intentional? Shouldn't there be `tf.stop_gradient` wrapped around the `tf.exp(log_alpha)` [here](https://github.com/rail-berkeley/softlearning/blob/master/softlearning/algorithms/sac.py#L269).

Thanks, 

Lukas

**Comment by jendelel** [comment]
After further inspection: In the critic, `self._alpha` is actually goes through the `tf.stop_gradient` later. It's only in the actor where is doesn't.

**Comment by hartikainen** [comment]
Good question. The reason we don't have to use `tf.stop_gradient` for alpha in the actor loss is that we explicitly set the `var_list` to only include the policy variables in the loss computation: https://github.com/rail-berkeley/softlearning/blob/6f25c33d904cf31911edee142d2b1247a3eeba7b/softlearning/algorithms/sac.py#L319.

This way alpha doesn't get updated.

Looking the critic update closer, it seems like the `tf.stop_gradient` is nowadays unnecessary since we excplicitly train only the `self._Qs`s trainable variables and not the target's or alpha's that are present in the Q target computation.
